pkg/machine/gvproxy.go

```diff
@@ -0,0 +1,31 @@
+package machine
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+
+	"github.com/sirupsen/logrus"
+)
+
+func CleanupGVProxy(f VMFile) error {
+	gvPid, err := f.Read()
+	if err != nil {
+		return fmt.Errorf("unable to read gvproxy pid file %s: %v", f.GetPath(), err)
+	}
+	proxyPid, err := strconv.Atoi(string(gvPid))
+	if err != nil {
+		return fmt.Errorf("unable to convert pid to integer: %v", err)
+	}
+	proxyProc, err := os.FindProcess(proxyPid)
+	if proxyProc == nil && err != nil {
+		return fmt.Errorf("unable to find process: %v", err)
+	}
+	if err := proxyProc.Kill(); err != nil {
+		logrus.Errorf("unable to kill gvproxy: %v", err)
+	}
+	if _, err := proxyProc.Wait(); err != nil {
+		logrus.Error(err)
+	}
```

Comment:
I don't think wait() here makes any sense, wait() will call the syscall wait() family so in order for that to function gvproxy must be a child process. However given the gvproxy is started with podman machine start and we call this from podman machine stop so it will never be our child and just log bogus errors without any context as well.

I am not sure where you tested this maybe it works differently on windows but on unix systems this should always fail